### ISSUE TITLE
Add "normal" method for adding source files

### ISSUE THREAD

**Issue by heinrichreimer** [DanySK/gradle-latex#2]
Currently we can only add source files by using the `String.invoke` extension function.
I would appreciate, if there would be a normal method `LatexExtension.add(name, ...)`, as I think that makes the code more readable.

**Comment by heinrichreimer** [comment]
We could also possibly implement `org.gradle.api.NamedDomainObjectContainer<LatexArtifact>`.
Might make the whole LaTeX configuration more familiar, as most Gradle users are already used to define their tasks in that manner.

**Comment by DanySK** [comment]
Can you provide some minimal configurations that you would like to be able to write?

e.g. I imagine you'd like something like:

```kotlin
latex {
    add("main.tex")
}
```

correct?

**Comment by heinrichreimer** [comment]
Yes. Especially, if we don't configure the artifact, it looks odd:
```kotlin
latex {
    "sample"()
}
```
Using `NamedDomainObjectContainer<LatexArtifact>`, it not only looks cleaner, but we also can configure artifacts lazy.
```kotlin
latex {
    register("sample")
    register("sample2") {
        // Won't be executed, until we look up artifacts.
    }
}
```
